src/proc/typifier.rs

```diff
@@ -50,6 +51,14 @@ pub enum ResolveError {
     FunctionReturnsVoid,
     #[error("Type is not found in the given immutable arena")]
     TypeNotFound,
+    #[error("Incompatible operand: {op} {operand}")]
+    IncompatibleOperand { op: String, operand: String },
+    #[error("Incompatible operands: {left} {op:?} {right}")]
+    IncompatibleOperands {
+        op: crate::Expression,
```

Comment:
I actually meant the left/right/operand to be `Expression`, not the `op`. I guess `op` works too

Reply:
Yeah sorry, my bad ... already reverted

Reply:
Hmm ... at this point in the code it only seems to have `Handle<Expression>` and no way to look it up in the `Arena<Expression>` - or am I missing something?

Reply:
@kvark if you have the time, maybe easier for you to do this edit - than me making a mess 😄

Reply:
ok, will do. Sorry about all the change requests :sweat_smile:

Reply:
np, just not too familier with this code, so could take quite some iteration before I got it right 😄

Reply:
I took a few attempts and wasn't able to find a nice way to expose this. Let's proceed!